src/DataFramesMeta.jl

```diff
@@ -583,6 +584,61 @@ macro transform(x, args...)
     esc(transform_helper(x, args...))
 end
 
+##############################################################################
+##
+## transform! & @transform!
+##
+##############################################################################
+
+
+function transform!_helper(x, args...)
+
+    t = (fun_to_vec(arg) for arg in args)
+
+    quote
+        $DataFrames.transform!($x, $(t...))
+    end
+end
+
+"""
+    @transform!(d, i...)
+
+Mutate `d` inplace to add additional columns or keys based on keyword arguments and return it. 
+
+### Arguments
+
+* `d` : an `AbstractDataFrame`, or `GroupedDataFrame`
+* `i...` : keyword arguments defining new columns or keys
+
+### Returns
+
+* `::DataFrame`
+
+### Examples
+
+```jldoctest
+julia> using DataFramesMeta
+
+julia> df = DataFrame(A = 1:3, B = [2, 1, 2]);
+
+julia> df2 = @transform!(df, a = 2 * :A, x = :A .+ :B);
+
+julia> (df === df2) && df2
```

Comment:
Can you just check that `df2.A === df.A`? Checking the full data frame seems unnecessary.